### PR TITLE
Do not use sles15sp1o

### PR DESCRIPTION
**Pull request by calancha**
## What does this PR change?

This official image is causing problems: we are not able to
deploy when this image is used in the suse clients.

### CODE REVIEW

**Comment by srbarrios** [comment]
I think we need to change it here too:
https://github.com/uyuni-project/sumaform/blob/8529d14433458bae1e6865e4cebf11f85a009ec8/backend_modules/libvirt/host/user_data.yaml#L142

And review this file! Because mayeb the if statement must contain both images in the comparison

**Comment by moio** [comment]
What is the problem exactly?

File: modules/server/main.tf [hunk 1]
```diff
@@ -2,8 +2,8 @@ variable "images" {
   default = {
```
Comment (srbarrios) [inline modules/server/main.tf:2]: I guess you can add the commented of the WORKAROUND or whatever, somewhere neat to that

**Comment by srbarrios** [comment]
> What is the problem exactly?

We don't exactly know it, Ricardo and us are trying to figure out, you can join the party in #galaxy-qa-review-40 or #galaxy-qam-review :) 
The thing is that is failing with this error, when using this image:
```
13:47:33  Error: Error: couldn't retrieve IP address of domain.Please check following: 
13:47:33  1) is the domain running proplerly? 
13:47:33  2) has the network interface an IP address? 
13:47:33  3) Networking issues on your libvirt setup? 
13:47:33   4) is DHCP enabled on this Domain's network? 
13:47:33  5) if you use bridge network, the domain should have the pkg qemu-agent installed 
13:47:33  IMPORTANT: This error is not a terraform libvirt-provider error, but an error caused by your KVM/libvirt infrastructure configuration/setup 
13:47:33   timeout while waiting for state to become 'all-addresses-obtained' (last state: 'waiting-addresses', timeout: 5m0s)
13:47:33  
13:47:33    on /home/jenkins/jenkins-build/workspace/manager-4.0-qam-setup-cucumber/results/sumaform/backend_modules/libvirt/host/main.tf line 65, in resource "libvirt_domain" "domain":
13:47:33    65: resource "libvirt_domain" "domain" {
```

**Comment by srbarrios** [comment]
> I'd recommend having this merged only as a last resource. Fixing the root cause would be preferable.
> 
> In the mean time running testsuites from this branch is feasible as a temporary workaround.

@moio I would strongly prefer to revert the default images, to use the images working, not breaking master.
Did you consider that some external people must not be aware of this issue and our master is broken for them?

It happened in the past, and we started bad moves, creating tags and branches that took too long, @Bischoff will agree on that.
Better to revert all the default images changed on the last commit, then work in a branch to assure that all the new default entries work in all our scenarios, then when all is good and safe we merge to master.

At that moment in QAM environment, sles15sp1o and opensuse150o break the deployment stage.

**Comment by moio** [comment]
>  Did you consider that some external people must not be aware of this issue and our master is broken for them?

I just tried again and, in my setup, deployment of sles15sp1o, opensuse150o and centos6o all work as expected.

I will need your help tomorrow to reproduce the problem reliably/safely and then continue from there.

**Comment by srbarrios** [comment]
> > Did you consider that some external people must not be aware of this issue and our master is broken for them?
> 
> I just tried again and, in my setup, deployment of sles15sp1o, opensuse150o and centos6o all work as expected.
> 
> I will need your help tomorrow to reproduce the problem reliably/safely and then continue from there.

This is the main.tf I was using before reverting to custom images. Jenkins history already erased the logs shown with this main.tf
@rjmateus is aware about them. 

In the controller for instance I had:
``` [2020-08-14T12:36:00.858Z] [0m[0mmodule.controller.module.controller.module.host.null_resource.provisioning[0] (remote-exec):     [0;31m Comment: Swap /extra_swapfile failed to activate. Added new entry to the fstab.[0;0m
```
```[2020-08-14T12:36:00.890Z] [0m[0mmodule.controller.module.controller.module.host.null_resource.provisioning[0] (remote-exec):     [0;31m Comment: Desired working directory "/root/spacewalk/testsuite" is not available[0;0m```

[SUSEManager-4.0-qam.tf.txt](https://github.com/uyuni-project/sumaform/files/5088797/SUSEManager-4.0-qam.tf.txt)

**Comment by srbarrios** [comment]
> > Did you consider that some external people must not be aware of this issue and our master is broken for them?
> 
> I just tried again and, in my setup, deployment of sles15sp1o, opensuse150o and centos6o all work as expected.
> 
> I will need your help tomorrow to reproduce the problem reliably/safely and then continue from there.

This is the main.tf I was using before reverting to custom images. Jenkins history already erased the logs shown with this main.tf
@rjmateus is aware about them. 

In the controller for instance I had:
``` 
[2020-08-14T12:36:00.858Z] [0m[0mmodule.controller.module.controller.module.host.null_resource.provisioning[0] (remote-exec):     [0;31m Comment: Swap /extra_swapfile failed to activate. Added new entry to the fstab.[0;0m
....
[2020-08-14T12:36:00.890Z] [0m[0mmodule.controller.module.controller.module.host.null_resource.provisioning[0] (remote-exec):     [0;31m Comment: Desired working directory "/root/spacewalk/testsuite" is not available[0;0m
```

Be sure to use those official images in those modules: server, proxy, controller

[SUSEManager-4.0-qam.tf.txt](https://github.com/uyuni-project/sumaform/files/5088869/SUSEManager-4.0-qam.tf.txt)

**Comment by calancha** [comment]
> @moio I would strongly prefer to revert the default images, to use the images working, not breaking master.
> Did you consider that some external people must not be aware of this issue and our master is broken for them?

SUSE is a open source company.  Sumaform is open source.
There is no reason to hide this issue.  Honesty is in our core values, and make the others trust on us ;-)

**Comment by juliogonzalez** [comment]
> SUSE is a open source company. Sumaform is open source.
> There is no reason to hide this issue. Honesty is in our core values, and make the others trust on us ;-)

I'd say it's not about hiding. Think about our developers. As I said at the chat, if they start using master and find it's broken, I will be one of the persons they will ping to ask what's going on.

`master` should just work.

We can make this public and visible with an issue for SLE (if there's an issue there, as it seems), and another one on this repository to use the official images again when SLE is fixed.

**Comment by calancha** [comment]
> `master` should just work.
> 
> We can make this public and visible with an issue for SLE (if there's an issue there, as it seems), and another one on this repository to use the official images again when SLE is fixed.

I agree with your 2 points: I also want master stable, and ping SLE people if there is something to fix there.
concerning this thread, there is no evidence that my changes are affecting test suite deployments (which is what I
have extensively tested).
Please, if someone thinks that these changes prevent sumaform deployment, show evidence, not guesses or feelings.
I have been deploying with this branch the whole last week with no issues.

**Comment by moio** [comment]
No worries @calancha, I'll be looking into @srbarrios's specific configuration today and I hope to be able to come out with an exact explanation, then I will report here.

**Comment by Bischoff** [comment]
It smells like qemu-agent not installed in cloud init. I agree with @moio this should be a last resort only. I'd vote for closing this PR only if we fail debugging that.

**Comment by moio** [comment]
In my understanding this is addressed by #767, can you confirm @Bischoff?

**Comment by Bischoff** [comment]
> In my understanding this is addressed by #767, can you confirm @Bischoff?

My PR #767 removes the deployment problems when sles15sp1o is used. However, there could be other problems linked to sles15sp1o in secondary features phase. I just don't know, it would need investigation.

To give you a taste of it:
* [https://ci.suse.de/view/Manager/view/Manager-4.0/job/manager-4.0-cucumber-PRV/541/#showFailuresLink](4.0 Provo), with "sumaform" images, 4 failures
* [https://ci.suse.de/view/Manager/view/Manager-4.0/job/manager-4.0-cucumber-nue/108/#showFailuresLink](4.0 Nuremberg), with "official" images, 41 failures

In the second case, many failures are linked to CentOS or to Ubuntu, but some might be on sles15sp1.

**Comment by moio** [comment]
Let's revisit this after Friday then

**Comment by srbarrios** [comment]
@calancha we don't need anymore this PR isn't it?

**Comment by calancha** [comment]
Closing out-of-date PR (we are already using sles15sp1o successfully).

**Pull request closed without merging**